src/psyclone/transformations.py

```diff
@@ -666,20 +799,22 @@ def _directive(self, children, collapse=None):
         :param children: list of Nodes that will be the children of \
                          the created directive.
         :type children: list of :py:class:`psyclone.psyir.nodes.Node`
-        :param int collapse: currently un-used but required to keep \
-                             interface the same as in base class.
+        :param int collapse: number of nested loops to collapse or None if \
+                             no collapse attribute is required.
+
         :returns: the new node representing the directive in the AST
-        :rtype: :py:class:`psyclone.psyir.nodes.OMPDoDirective`
-        :raises NotImplementedError: if a collapse argument is supplied.
+        :rtype: :py:class:`psyclone.psyir.nodes.OMPDoDirective` or \
+                :py:class:`psyclone.psyir.nodes.OMPLoopDirective`
+
         '''
-        # TODO 1370: OpenMP loop functions don't support collapse
-        if collapse:
-            raise NotImplementedError(
-                "The COLLAPSE clause is not yet supported for '!$omp do' "
-                "directives.")
-        _directive = OMPDoDirective(children=children,
-                                    omp_schedule=self.omp_schedule,
-                                    reprod=self._reprod)
+        if self._omp_worksharing:
+            _directive = OMPDoDirective(children=children,
```

Comment:
Please could you add a comment here explaining that we don't support collapse for worksharing directives because the `gen` code-generation pathway doesn't support it. I think that means keeping the TODO referencing #1370.

Reply:
Done (I haven't specified that is a gen issue because after another look the backend also doesn't do it, it may be a matter of just updating the begin_string, but since there is already the issue for this I will leave the TODO in this PR)